src/PokemonGymImpl.java

```diff
@@ -261,4 +267,28 @@ public void attackOrChange(Pokemon pokemon, Pokemon gymPokemon, PokemonTrainer t
         }
     }
 
+    private void feedPokemon(Pokemon pokemon, PokemonTrainer trainer) {
+        String choosenFood = foodList(trainer.getFoods()).toLowerCase(Locale.ROOT);
+        System.out.println("You have fed "+ pokemon.getName() + " an " +choosenFood);
+
+        for (Food food : trainer.getFoods()) {
+            if (choosenFood.equals(food.getName().toLowerCase(Locale.ROOT))) {
+                pokemon.setHp(pokemon.getHp() + food.getHealthPoints());
+                System.out.println(food.getHealthPoints() + "Hp restored");
+            }
+        }
+    }
+
+    private String foodList(List<Food> foods) {
+        Scanner speler_A = new Scanner(System.in);
+        List<String> foodNames = new ArrayList<>();
+        for (Food food : foods) {
+            foodNames.add(food.getName());
```

Comment:
Deze loop is enkel om de Food naar een String om te zetten en is een prima oplossing. Maar je kunt hier ook heel makkelijk gebruik maken van de Stream API. Je krijgt dan: `List<String> foodNames = foods.stream().map(food->food.getName()).toList();`.